### PR TITLE
source-freshbooks contribution from bishalbera

### LINKED ISSUE #47421: [Connector-Request] FreshBooks
### Topic

This is a suggestion to create a connector for FreshBooks using the UI Connector Builder. The website for this is https://www.freshbooks.com/

### Relevant information

_No response_

### PR DESCRIPTION
**Pull request by bishalbera**
## What
Closes #47421 

This PR adds a new connector for source FreshBooks (source-freshbooks).

Contributor-provided connector description:

FreshBooks connector  seamlessly syncs invoicing, expenses, and client data from FreshBooks into data warehouses or analytics platforms. 
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `redirect_uri` | `string` | Redirect Uri.  |  |
| `account_id` | `string` | Account Id.  |  |
| `client_refresh_token` | `string` | Refresh token.  |  |
| `oauth_access_token` | `string` | Access token. The current access token. This field might be overridden by the connector based on the token refresh endpoint response. |  |
| `oauth_token_expiry_date` | `string` | Token expiry date. The date the current access token expires in. This field might be overridden by the connector based on the token refresh endpoint response. |  |
| `business_uuid` | `string` | Business uuid.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| user |  | DefaultPaginator | ✅ |  ❌  |
| clients |  | DefaultPaginator | ✅ |  ❌  |
| invoices | id.invoiceid | DefaultPaginator | ✅ |  ❌  |
| expenses | id.expenseid | DefaultPaginator | ✅ |  ❌  |
| expense_summaries |  | DefaultPaginator | ✅ |  ❌  |
| expense_categories | id | DefaultPaginator | ✅ |  ❌  |
| invoice_details | invoiceid | DefaultPaginator | ✅ |  ❌  |
| expense_details | expenseid | DefaultPaginator | ✅ |  ❌  |
| accounts |  | DefaultPaginator | ✅ |  ❌  |
| taxes | taxid | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: docs/integrations/sources/freshbooks.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `client_id` | `string` | Client ID.  |  |
+| `client_secret` | `string` | Client secret.  |  |
+| `redirect_uri` | `string` | Redirect Uri.  |  |
+| `account_id` | `string` | Account Id.  |  |
+| `client_refresh_token` | `string` | Refresh token.  |  |
+| `oauth_access_token` | `string` | Access token. The current access token. This field might be overridden by the connector based on the token refresh endpoint response. |  |
+| `oauth_token_expiry_date` | `string` | Token expiry date. The date the current access token expires in. This field might be overridden by the connector based on the token refresh endpoint response. |  |
+| `business_uuid` | `string` | Business uuid.  |  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/freshbooks.md:15]: Provide information how to find Account ID and Business UUID

File: docs/integrations/sources/freshbooks.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# FreshBooks
+FreshBooks connector  seamlessly syncs invoicing, expenses, and client data from FreshBooks into data warehouses or analytics platforms. 
+
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `client_id` | `string` | Client ID.  |  |
+| `client_secret` | `string` | Client secret.  |  |
+| `redirect_uri` | `string` | Redirect Uri.  |  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/freshbooks.md:10]: Provide more information about Redirect URI, it need to be a variable?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Connector Looks good, left couple of comments about documentation

**Comment by marcosmarxm** [comment]
Ping me when this is done.

File: docs/integrations/sources/freshbooks.md (hunk 2 above)
Comment (bishalbera) [reply docs/integrations/sources/freshbooks.md:10]: @marcosmarxm yes as it states here in the pic
Screenshot 2024-11-07 at 9 25 22 AM

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**